toolkits/jira/arcade_jira/tools/sprint_planning.py

```diff
@@ -0,0 +1,410 @@
+            error_entry = create_error_entry(
+                board_id,
+                f"Board '{board_id}' could not be resolved",
+            )
+            results["errors"].append(error_entry)
+            return
+
+        # Check if board supports sprints and get final type, fetch sprints in one call
+        params = build_sprint_params(offset, sprints_per_board, state)
+        supports_sprints, final_type, response = await _try_fetch_sprints_and_determine_type(
+            client, board_info, cast(dict[str, Any], params)
+        )
+
+        if not supports_sprints:
+            error_entry = create_error_entry(
+                board_id,
+                f"Board '{board_info.get('name', board_id)}' does not support sprints "
+                f"(type: {board_info.get('type', 'unknown')}). "
+                f"Only Scrum boards support sprints.",
+                board_info.get("name", "Unknown"),
+                board_id_resolved,
+            )
+            results["errors"].append(error_entry)
+            return
+
+        # Update board type if it changed (simple -> scrum)
+        board_info["type"] = final_type
+
+        # Process the sprints we already fetched
+        if response is None:
+            # This should not happen if supports_sprints is True, but handle it gracefully
+            error_entry = create_error_entry(
+                board_id,
+                f"Unexpected error: No sprint data received for board "
+                f"'{board_info.get('name', board_id)}'",
+                board_info.get("name", "Unknown"),
+                board_id_resolved,
+            )
+            results["errors"].append(error_entry)
+            return
+
+        sprints = [clean_sprint_dict(s) for s in response.get("values", [])]
+
+        # Apply date filtering if specified
+        if start_date or end_date or specific_date:
+            sprints = _filter_sprints_by_date(sprints, start_date, end_date, specific_date)
+
+        # Sort sprints with latest first (by end date, then start date, then ID)
+        sprints = _sort_sprints_latest_first(sprints)
+
+        results["boards"].append(board_info)
+        results["sprints_by_board"][board_id_resolved] = create_sprint_result_dict(
+            board_info, sprints, response
+        )
+
+    except ToolExecutionError:
+        # Re-raise ToolExecutionErrors as-is
+        raise
+    except Exception as e:
+        error_entry = create_error_entry(
+            board_id,
+            f"Unexpected error processing board '{board_id}': {e!s}",
+        )
+        results["errors"].append(error_entry)
+
+
+async def _try_fetch_sprints_and_determine_type(
+    client: JiraClient, board_info: dict[str, Any], params: dict[str, Any]
+) -> tuple[bool, str, dict[str, Any] | None]:
+    """
+    Try to fetch sprints for a board and determine if it supports sprints.
+    Returns (supports_sprints, final_board_type, response).
+    """
```

Comment:
typically we try to have 
```
"""
    Args:
        arg_name: arg_description
    Returns:
        return_description
    Raises:
        exception_type: when_this_is_raised_description
"""
```
in the docstring for all non-tools that are complex enough.